activerecord/test/cases/persistence_test.rb

```diff
@@ -168,6 +169,14 @@ def test_becomes_includes_changed_attributes
     assert_equal %w{name}, client.changed
   end
 
+  def test_becomes_includes_association_cache
+    assert_not_equal Company.count, 0
+    Company.preload(:contracts).each do |company|
+      client = company.becomes(Client)
+      assert client.association(:contracts).loaded?, "client contracts should be loaded"
```

Comment:
Could we test this by checking the number of queries performed, rather than reaching into the internals of the object?